nxdrive/direct_edit.py

```diff
@@ -308,18 +308,35 @@ def _download(
 
         if not pair:
             if url:
+                max_retry = 3
+                tmp_file = None
                 try:
-                    engine.remote.download(
-                        quote(url, safe="/:"),
-                        file_path,
-                        file_out,
-                        blob.digest,
-                        callback=self.stop_client,
-                        is_direct_edit=True,
-                        engine_uid=engine.uid,
-                    )
+                    for try_count in range(max_retry):
+                        try:
+                            tmp_file = engine.remote.download(
+                                quote(url, safe="/:"),
+                                file_path,
+                                file_out,
+                                blob.digest,
+                                callback=self.stop_client,
+                                is_direct_edit=True,
+                                engine_uid=engine.uid,
+                            )
+                            break
+                        except CorruptedFile:
+                            description = "Unable to perform DirectEdit because of corrupted download\nRetrying..."
+                            log.warning(description, exc_info=True)
+                            self.directEditError.emit(
+                                "DIRECT_EDIT_CORRUPTED_DOWNLOAD", []
+                            )
+                            delay = 5 * (try_count + 1)
+                            sleep(delay)
+                            continue
```

Comment:
You can remove that line.

And perhaps is it a good place to use the `for/else` idiom:
```python
for try_count in range(max_retry):
    ...
else:
    self.directEditError.emit("DIRECT_EDIT_FAILURE", [])
    return None
```
The `else` will be taken only if the `break` in the loop is not taken.